### ISSUE TITLE
Please add Nagios output

### ISSUE THREAD

**Issue by xoroz** [Qarj/WebInject#11]
First of all, thank you so much for reviving the webinject.pl and adding so many cool features!!! 

Please add the option to have the nagios output, it should be a single line output
This is the <reporttype>nagios</reporttype> option on original webinject project
http://www.webinject.org/manual.html

Example:
./check_webinject.pl -c confs/accerta_cfg.xml
WebInject OK - All tests passed successfully in 0.927 seconds |time=0.927;;;0

Thank you so much!

**Comment by Qarj** [comment]
Hi, thanks for checking out the project!

I've never used nagios which is why I removed it. However it looks pretty simple to support, I can add it back in if you are willing to test it for me :)

**Comment by xoroz** [comment]
Sure will test it and will also make people know it exists.
I will make you famous :)
Let me know when its ready.

**Comment by Qarj** [comment]
Hi @xoroz I've now added Nagios output back into WebInject and have created some self tests for this.

I've tested this on Fedora 23 and Windows and it seems like it is working. However if you can give it a go and let me know, that would be great since I have no experience with Nagios.

Note that the timing values reported will be the sum of the response times for each test step. This is a change from WebInject 1.41 which reported the total run time. The difference is slight, however the sum of the response times is a more accurate value since it will not include most of the WebInject execution time (e.g. compiling packages, parsing, logging and verification time).

When everything is OK, I'll update the manual and bump the version number :)

**Comment by xoroz** [comment]
Sorry but did not work for me....
Maybe you can tell me how you configured yours..

I did like these
accerta.xml
<testcasefile>accerta_cfg.xml</testcasefile>
<reporttype>nagios</reporttype>

./webinject.pl -c accerta.xml
accerta.xml:2: parser error : Extra content at the end of the document
<reporttype>nagios</reporttype>
^

./webinject.pl -v

WebInject version 1.99
For more info: https://github.com/Qarj/WebInject

So its not taking the reporttype .. how did you configure it?

**Comment by Qarj** [comment]
If accerta_cfg.xml is the config file, then you need to kick it off like this:

```
./webinject.pl -c accerta_cfg.xml
```

The contents of your config file should point to accerta.xml and also indicate a reporttype of nagios. For example:

#### accerta_cfg.xml

```
<root>
    <testcasefile>accerta.xml</testcasefile>
    <reporttype>nagios</reporttype>
    <globaltimeout>100</globaltimeout>
</root>
```

and your testcasefile needs to be in regular format. Here is a simple example:

#### accerta.xml

```
<testcases repeat="1">

<case
    id="10"
    description1="Get Totaljobs Home Page"
    description2="Description 2 is used to display additional information about the test step at hand"
    method="get"
    url="http://www.totaljobs.com"
    verifypositive="recommended"
/>

<case
    id="20"
    description1="Get Totaljobs Account Register Page"
    description2="Runs over https"
    method="get"
    url="https://www.totaljobs.com/account/register"
    verifypositive="mandatory"
/>

</testcases>
```

**Comment by xoroz** [comment]
It worked!
I was missing the root tag.

Cheers!!!

Il 22/Lug/2016 10:26, "Tim Buckland" notifications@github.com ha scritto:

**Comment by Qarj** [comment]
Great news!

Yeah the fact that xml requires some form of root tag is a real pain. Glad it works though :)

**Comment by Qarj** [comment]
Implemented in ver 2.0.0
